### DISCUSSION TITLE
Welcome to Topaz Discussions!

### DISCUSSION THREAD

**Discussion by koculu** [koculu/Topaz#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by sachinpatel2193** [comment]
Hi Koculu,
I am trying to use Topaz in one of the dotnet project. My script is returning a response object. I checked the methods for Executing scripts, the methods has void return type. Is there an alternative option to execute the script that returns something?

**Reply by koculu** [reply]
You can use ExecuteExpression or InvokeFunction methods.

**Comment by blakepell** [comment]
I just wanted to say, I absolutely love this project and have used it in a lot of places for my personal projects.  I thought I'd share some of the things I've used it for in case you were wondering what people use it for.

My passion project is a MUD client written in WPF (allows players to play online multiplayer text based games that have been around since the 90's).  It was my first WPF project and it works fine, but I made a fundamental missteps.  I'm rewriting it from the ground up currently and replacing MoonSharp with Topaz.  As part of that I've written/am working on a code editor for Topaz using AvalonEdit that does syntax highlighting and some intellisense from things registered with Topaz (you can see here objects I've loaded into Topaz).

![image](https://github.com/koculu/Topaz/assets/5958430/acf648cf-f6b7-4d46-87bc-f29b3a16ef80)

![image](https://github.com/koculu/Topaz/assets/5958430/57a5b658-a12f-4a28-a7a0-90f8296620f8)

I have a program I wrote to manage my music playlists.  I use Topaz in it which allows me to write scripts to iterate over all the songs in a playlist and change the metadata tags about the MP3 files.  I have a text based command launcher I'm almost done with that can execute topaz scripts as one of it's outputs (it starts processes, shells folders, searches files and can run topaz scripts as essentially macros).

<img width="761" alt="image" src="https://github.com/koculu/Topaz/assets/5958430/8a529ef0-1eb2-4afe-bc2d-d668c456174f">

One more I did this morning when I woke up (too) early, I have a WPF control that's a generic list view, that will render any list without having to write XAML.  For the cases where I need to allow a property to do something that's not a default value I use attributes I put on the property that contain topaz scripts.  So in the below, my value is represented by a progress bar in the grid.  But, it needed to have it's height changed to be visible and it needed a max value set on it.  You can see, I have an OnLoaded property that Topaz then handles when that column's data template is created.

<img width="561" alt="image" src="https://github.com/koculu/Topaz/assets/5958430/0286d40b-7c32-434d-8d22-393d1f6fcc93">

**Reply by koculu** [reply]
Thank you for sharing this. Awesome.
